### ISSUE TITLE
Add Canary Build as Beta or Edge

### ISSUE THREAD

**Issue by RickAndTired** [snapcrafters/discord#81]
Discord has a Canary build they use for testing

I think it would be beneficial to add this build as a Beta or Edge channel in the Snap

I've seen a few people have issues with the current build of Discord and the Canary build solving their problems. Also it will help users test new builds.

Deb
https://discordapp.com/api/download/canary?platform=linux
Tar.gz
https://discordapp.com/api/download/canary?platform=linux&format=tar.gz

**Comment by devksingh4** [comment]
I have published at https://snapcraft.io/discord-canary and plan to make a PR to make this an official snapcrafters project.

**Comment by Fuseteam** [comment]
@devksingh4 isn't the idea to add it as a channel to this particular snap?

**Comment by devksingh4** [comment]
For me and my use case, it makes sense for me to manage them as separate snaps since I want to have both versions installed simulaneously. That is why I have made them two separate snaps.

**Comment by Fuseteam** [comment]
@devksingh4 yeah that's not a bad idea, but uh do happen to know what would be needed to add it as a channel to this snap anyway? that way we would have both as an option

**Comment by devksingh4** [comment]
The snapcraft.yml from my repository can be used to build a snap, and then promote it to edge and not to candidate or stable. It is easy for them to do, I just don't know how it works with the automation i presume is being used to build these. If different snapcraft.yml files can be used to build different channels, then this can work perfectly.

**Comment by Fuseteam** [comment]
> The snapcraft.yml from my repository can be used to build a snap, and then promote it to edge and not to candidate or stable. It is easy for them to do, I just don't know how it works with the automation i presume is being used to build these. If different snapcraft.yml files can be used to build different channels, then this can work perfectly.

ah cool sounds like we need to get input from the maintainer

**Comment by merlijn-sebrechts** [comment]
With the current automation, we can actually create an `edge` branch that published to the `edge` channel in the Snap Store. Feel free to take a look at the automation in the `.github` folder to see how it works.
